### PR TITLE
Add NotCheckpointSafe annotations & JPP support for generated source

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Annotated `java.lang.ClassValue.initializeMap()`, `java.lang.invoke.ClassSpecializer.findSpecies()`, and `java.util.concurrent.ConcurrentHashMap.computeIfAbsent()` with `NotCheckpointSafe` to prevent `JVMCheckpointException`/deadlock;
Added `CRIU` `JPP` flag at `StandardCharsets.java.template`;
Modified `GensrcCharsetMapping.gmk` to run JPP after `StandardCharsets.java` generation.

Cherry-pick https://github.com/ibmruntimes/openj9-openjdk-jdk17/pull/128

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

File: closed/GensrcJ9JCL.gmk [hunk 1]
```diff
@@ -51,8 +51,11 @@ $(eval $(call SetupCopyFiles,COPY_OVERLAY_FILES, \
 	SRC := $(TOPDIR), \
 	DEST := $(SUPPORT_OUTPUTDIR)/overlay, \
 	FILES := \
+		src/java.base/share/classes/java/lang/ClassValue.java \
+		src/java.base/share/classes/java/lang/invoke/ClassSpecializer.java \
 		src/java.base/share/classes/java/security/Security.java \
 		src/java.base/share/classes/java/util/Timer.java \
+		src/java.base/share/classes/java/util/concurrent/ConcurrentHashMap.java \
```
Comment (keithc-ca) [inline closed/GensrcJ9JCL.gmk:58]: Please use the same order as in https://github.com/ibmruntimes/openj9-openjdk-jdk17/pull/128.

File: src/java.base/share/classes/java/lang/ClassValue.java [hunk 2]
```diff
@@ -34,6 +40,10 @@
 import static java.lang.ClassValue.ClassValueMap.probeHomeLocation;
 import static java.lang.ClassValue.ClassValueMap.probeBackupLocations;
 
+/*[IF CRIU_SUPPORT]*/
+import openj9.internal.criu.NotCheckpointSafe;
+/*[ENDIF] CRIU_SUPPORT*/
```
Comment (keithc-ca) [inline src/java.base/share/classes/java/lang/ClassValue.java:45]: Please add spaces like in https://github.com/ibmruntimes/openj9-openjdk-jdk17/pull/128.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: closed/GensrcJ9JCL.gmk (hunk 1 above)
Comment (JasonFengJ9) [reply closed/GensrcJ9JCL.gmk]: Redo cherry-pick from https://github.com/ibmruntimes/openj9-openjdk-jdk17/pull/128

File: src/java.base/share/classes/java/lang/ClassValue.java (hunk 2 above)
Comment (JasonFengJ9) [reply src/java.base/share/classes/java/lang/ClassValue.java]: Cherry-pick https://github.com/ibmruntimes/openj9-openjdk-jdk17/pull/128

**Comment by JasonFengJ9** [comment]
Jenkins test sanity.functional zlinux jdknext

**Comment by keithc-ca** [comment]
The test failures should hopefully be addressed by https://github.com/eclipse-openj9/openj9/pull/15625.

**Review by keithc-ca — APPROVED** [review]

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
THe test failure is unrelated to these changes

```
===============================================
Running test cmdLineTester_jvmtitests_Java11andUp_0 ...
===============================================
cmdLineTester_jvmtitests_Java11andUp_0 Start Time: Tue Jul 26 18:36:42 2022 Epoch Time (ms): 1658860602570
variation: Mode110
JVM_OPTIONS:  -Xjit -Xgcpolicy:gencon -Xnocompressedrefs 
java.io.FileNotFoundException: /home/jenkins/workspace/Test_openjdknext_j9_sanity.functional_s390x_linux_Personal_testList_1/aqa-tests/TKG/../../jvmtest/functional/cmdLineTests/jvmtitests/jvmtitests_excludes_20.xml (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:158)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:112)
	at com.oti.j9.exclude.ExcludeList.readFrom(ExcludeList.java:71)
	at MainTester.main(MainTester.java:97)
java.lang.NullPointerException
	at java.base/java.io.Reader.<init>(Reader.java:168)
	at java.base/java.io.InputStreamReader.<init>(InputStreamReader.java:89)
	at com.oti.j9.exclude.XMLParser.parse(XMLParser.java:468)
	at com.oti.j9.exclude.XMLInterface.parse(XMLInterface.java:138)
	at com.oti.j9.exclude.ExcludeList.readFrom(ExcludeList.java:83)
	at com.oti.j9.exclude.ExcludeList.readFrom(ExcludeList.java:75)
	at MainTester.main(MainTester.java:97)
```

**Pull request merged**